File: app/src/main/java/eu/darken/sdmse/common/forensics/csi/dalvik/tools/ExistCheck.java
//package eu.darken.sdmse.common.forensics.csi.dalvik;
//
//
//import java.util.Collection;
//
//import eu.thedarken.sdm.tools.forensics.OwnerInfo;
//import eu.thedarken.sdm.tools.forensics.csi.CSIHelper;
//import eu.thedarken.sdm.tools.forensics.csi.CSIProcessor;
//import eu.thedarken.sdm.tools.io.SDMFile;
//
//public class ExistCheck extends CSIHelper {
//    protected ExistCheck(CSIProcessor csiProcessor) {
//        super(csiProcessor);
//    }
//
//    public boolean check(OwnerInfo ownerInfo, Collection<SDMFile> candidates) {
//        for (SDMFile can : candidates) {
//            if (can.getJavaFile().exists()) {
//                ownerInfo.setUnknownOwner(true);
//                return true;
//            }
//        }
//        return false;
//    }
//}

File: app/src/main/java/eu/darken/sdmse/common/forensics/csi/dalvik/tools/ExistCheck.kt
package eu.darken.sdmse.common.forensics.csi.dalvik.tools

import dagger.Reusable
import eu.darken.sdmse.common.files.GatewaySwitch
import eu.darken.sdmse.common.files.exists
import eu.darken.sdmse.common.files.local.LocalPath
import eu.darken.sdmse.common.forensics.csi.dalvik.DalvikCheck
import javax.inject.Inject

@Reusable
class ExistCheck @Inject constructor(
    private val gatewaySwitch: GatewaySwitch
) : DalvikCheck {

    suspend fun check(candidates: Collection<LocalPath>): DalvikCheck.Result {
        return DalvikCheck.Result(
            hasKnownUnknownOwner = candidates.any {
                it.exists(gatewaySwitch)
            }
        )
    }
}